src/Physics/HadronTransport/INukeUtils2025.h

```diff
@@ -0,0 +1,123 @@
+//____________________________________________________________________________
+/*!
+
+\namespace genie::intranuke
+
+\brief     INTRANUKE utilities
+
+\author    Jim Dobson <j.dobson07 \at imperial.ac.uk>
+           Imperial College London
+
+           Costas Andreopoulos <c.andreopoulos \at cern.ch>
+           University of Liverpool
+
+	   Aaron Meyer <asm58 \at pitt.edu>
+	   Pittsburgh University
+
+\created   Mar 03, 2009
+
+\cpright   Copyright (c) 2003-2025, The GENIE Collaboration
+           For the full text of the license visit http://copyright.genie-mc.org
+
+*/
+//____________________________________________________________________________
+
+#ifndef _INTRANUKE_UTILS_2025_H_
+#define _INTRANUKE_UTILS_2025_H_
+
+#include <TGenPhaseSpace.h>
+
+#include "Physics/NuclearState/NuclearModelI.h"
+#include "Physics/HadronTransport/INukeHadroFates2025.h"
+#include "Physics/HadronTransport/INukeMode.h"
+#include "Physics/HadronTransport/INukeNucleonCorr.h"
+
+class TLorentzVector;
+
+namespace genie {
+
+class GHepRecord;
+class GHepParticle;
+class Intranuke2025;
+class PDGCodeList;
+
+namespace utils {
+namespace intranuke2025
+{
+  //! Hadron survival probability
+  double ProbSurvival(
+    int pdgc, const TLorentzVector & x4, const TLorentzVector & p4, double A,
+    double Z, double mfp_scale_factor, const Intranuke2025& fsi_model );
+
+  //! Mean free path (pions, nucleons)
+  double MeanFreePath(
+    int pdgc, const TLorentzVector & x4, const TLorentzVector & p4, double A,
+    double Z, double nRpi=0.5, double nRnuc=1.0, const bool useOset = false, const bool altOset = false, const bool xsecNNCorr = false, string INukeMode = "XX2025");
+
+  //! Mean free path (Delta++ **test**)
+  double MeanFreePath_Delta(
+			    int pdgc, const TLorentzVector & x4, const TLorentzVector & p4, double A );
+
+  //! Distance to exit
+  double Dist2Exit(
+    const TLorentzVector & x4, const TLorentzVector & p4,
+    double A, double NR=3, double R0=1.4);
+
+  //! Distance to exit
+  double Dist2ExitMFP(
+   int pdgc, const TLorentzVector & x4, const TLorentzVector & p4,
+    double A, double Z, double NR=3, double R0=1.4);
+
+  //! Step particle
+  void StepParticle(
+    GHepParticle * p, double step, double nuclear_radius=-1.);
+
+
+  //! Intranuke utility functions
+
+  bool TwoBodyCollision(
+    GHepRecord* ev, int pcode, int tcode, int scode, int s2code, double C3CM, GHepParticle* p,
+    GHepParticle* t, int &RemnA, int &RemnZ, TLorentzVector &RemnP4, EINukeMode mode=kIMdHA);
+
+  bool TwoBodyKinematics(
+    double M3, double M4, TLorentzVector tP1L, TLorentzVector tP2L,
+    TLorentzVector &tP3L, TLorentzVector &tP4L, double C3CM, TLorentzVector &RemnP4, double bindE=0);
+
+  bool ThreeBodyKinematics(
+    GHepRecord* ev, GHepParticle* p, int tcode, GHepParticle* s1, GHepParticle* s2, GHepParticle* s3,
+    bool DoFermi=false, double FermiFac=0, double FermiMomentum=0, const NuclearModelI* Nuclmodel=(const NuclearModelI*)0);
```

Comment:
Rather than casting `0` to `const NuclearModelI*`, I think it's more straightforward to use C++11 keyword `nullptr` which is type safe.